backend/es/views.py

```diff
@@ -20,33 +20,19 @@ def get_query_parameters(request):
             for key in request.query_params
         }
 
-
-def specify_tags(query, corpus_name, request):
-    '''
-    Specifies tag contents if needed.
-
-    If the query JSON contains a `tags` key, it is removed and replaced with
-    a filter on the tags' document IDs.
-    '''
-
-    tags = query.pop('tags', None)
-    return include_tag_filter(query, tags, corpus_name)
-
 class ForwardSearchView(APIView):
     '''
     Forward search request to elasticsearch
 
-    The request should specify a query in the elasticsearch query DSL; c.f.
-
+    The request may specify:
+    - `es_query`: a query in the elasticsearch query DSL; c.f.
     https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html
+    - `tags`: an array of tag IDs that the results should be filtered on. Multiple tags
+    are combined with OR logic: a document must match any of them.
 
-    The request data should be the query JSON. You can optionally include
-    extra key/value pairs as query parameters in the request. They
-    will be merged with the JSON data.
-
-    On top of any properties included in the query DSL, the request may also specify
-    a `tags`property. This should be an array of values, referring to IDs of tags
-    on which the results should be filtered.
+    You can optionally include extra key/value pairs as query parameters
```

Comment:
I think we should point out that these MUST be ES valid keys/values. Is this validated?

Reply:
Good point, I'll add it to the docstring. As for validation: elasticsearch obviously validates it so the request will fail if you add strange values. I don't think we need to add our own validation layer unless we want to block some queries that are allowed by elasticsearch.